File: AlphaMind-main/frontend/src/types/invoice.ts
export interface Invoice {
  id: number;
  user_id: number;
  invoice_number: string;
  total_searches: number;
  amount: number;
  invoice_date: string;
  pdf_path: string | null;
  is_paid: boolean;
}

export interface InvoiceGenerateResponse {
  invoice: Invoice;
  message: string;
}

File: AlphaMind-main/backend/app/schemas/invoice.py
from datetime import datetime
from pydantic import BaseModel
class InvoiceResponse(BaseModel):
    id: int
    user_id: int
    invoice_number: str
    total_searches: int
    amount: float
    invoice_date: datetime | None = None
    pdf_path: str | None = None
    is_paid: bool = False
    class Config:
        from_attributes = True
class InvoiceGenerateResponse(BaseModel):
    invoice: InvoiceResponse
    message: str = "Invoice generated successfully"